flink-yarn/src/main/java/org/apache/flink/yarn/AbstractYarnClusterDescriptor.java

```diff
@@ -988,7 +992,7 @@ public ApplicationReport startAppMaster(
 		final String customApplicationName = customName != null ? customName : applicationName;
 
 		appContext.setApplicationName(customApplicationName);
-		appContext.setApplicationType("Apache Flink");
+		appContext.setApplicationType(applicationType.isEmpty() ? "Apache Flink" : applicationType);
```

Comment:
Can't we just do the following?
  appContext.setApplicationType(dynProperties.getOrDefault("application-type",  "Apache Flink"));